src/App.tsx

```diff
@@ -15,47 +16,122 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((f1, f2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return f1.localeCompare(f2);
+      case SortType.LENGTH:
+        return f1.length - f2.length;
+      default:
+        return SortType.NONE;
+    }
+  });
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  // eslint-disable-next-line no-console
+  console.log(sortType, isReversed);
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setReverse] = useState(false);
+
+  const isResetButtonVisible = isReversed || sortType !== SortType.NONE;
+
+  const getSortAlphabet = () => {
```

Comment:
if your fuction name starts with 'get' - expected, that you get smth from this function
rename it, please, f.e.:
handleSortAlphabetycally
sortByAlphabet

fix al similar cases